### ISSUE TITLE
Add alt text to gallery image upload and edit

### ISSUE THREAD

**Issue by chronoDave** [modrinth/code#5797]
### Please confirm the following.

- [x] I checked the [existing issues](https://github.com/modrinth/code/issues?q=is%3Aissue) for duplicate feature requests
- [x] I have checked that this feature request is not on our [roadmap](https://roadmap.modrinth.com)

### What parts of Modrinth is your feature request related too?

Modrinth.com website

### Is your suggested feature related to a problem? Please describe.

There is currently no way for authors to edit the alt text used for the gallery images. It seems to default to the title of the image, which is not descriptive. Adding an options for authors to edit alt text would greatly benefits users relying on screen readers as the existing alt text is non-descriptive. If you're unfamiliar with alt text and how it benefits users, gov.uk has an excellent blog post with more context: https://accessibility.blog.gov.uk/2024/07/17/alternative-text-in-action/

### Describe the solution you'd like

An additional "Alternative text" field similiar to Description above Title would be sufficient. Preferably mandatory, with a link explaining what it is / how it benefits users, though making it optional and having the gallery defaulting to empty alt text (`alt=""`) would also be fine. No alt text is better than bad alt text.

### Describe alternatives you've considered

There are no alternatives as the platform, as it stands, does not allow authors to edit image alt text whatsoever.

### Additional context

User stories explaining how alt text helps them on the web: https://www.w3.org/WAI/fundamentals/accessibility-principles/#stories-related-to-text-alternatives
